### PR TITLE
A11Y: sets html lang using current locale and not default

### PR DESCRIPTION
**Pull request by jjaffeux**

### CODE REVIEW

**Comment by jjaffeux** [comment]
@gschlager do you see any issue with us doing this ?

Prior to this change we were always showing `<html lang="en" ...`

**Comment by CvX** [comment]
The spec failure seems related 😃

**Comment by jjaffeux** [comment]
@gschlager @CvX I might be missing some encoding knowledge here, but it seems that setting the I18n.locale changed the encoding of the mail to quoted-printable instead of 7bit and the matching used in the test was not working anymore.

Any hints? Confirmation?

File: lib/email/styles.rb [hunk 1]
```diff
@@ -180,7 +180,7 @@ def format_html
       correct_footer_style_hilight_first
       reset_tables
 
-      html_lang = SiteSetting.default_locale.sub("_", "-")
+      html_lang = I18n.locale.to_s.sub("_", "-")
       style('html', nil, lang: html_lang, 'xml:lang' => html_lang)
       style('body', "text-align:#{ Rtl.new(nil).enabled? ? 'right' : 'left' };")
       style('body', nil, dir: Rtl.new(nil).enabled? ? 'rtl' : 'ltr')
```
Comment (gschlager) [inline lib/email/styles.rb:186]: After this change `Rtl.new(nil).enabled?` might return a wrong value because it doesn't use `I18n.locale`

**Review by gschlager — COMMENTED** [review]
> @gschlager @CvX I might be missing some encoding knowledge here, but it seems that setting the I18n.locale changed the encoding of the mail to quoted-printable instead of 7bit and the matching used in the test was not working anymore.
> 
> Any hints? Confirmation?

I assume Rails changes the encoding as soon as a language isn't English anymore. Makes kinda sense. The chances are high that the email will contain non-ASCII characters.

**Comment by jjaffeux** [comment]
@gschlager I updated the PR, what do you think of this approach ?

**Review by gschlager — APPROVED** [review]

**Pull request merged**